freeipa/src/main/java/com/sequenceiq/freeipa/controller/FreeIpaV1Controller.java

```diff
@@ -485,4 +490,13 @@ public UsedSubnetsByEnvironmentResponse getUsedSubnetsByEnvironment(String envir
                 .stream().map(s -> new UsedSubnetWithResourceResponse(s.getName(), s.getSubnetId(), s.getResourceCrn(), s.getType()))
                 .collect(Collectors.toList()));
     }
+
+    @Override
+    @CheckPermissionByResourceCrn(action = EDIT_ENVIRONMENT)
+    public FlowIdentifier rotateSecretsByCrn(
+            @ValidCrn(resource = CrnResourceDescriptor.ENVIRONMENT) @ResourceCrn @NotEmpty @TenantAwareParam String environmentCrn,
+            @Valid @NotNull FreeIpaSecretRotationRequest request) {
+        String accountId = ThreadBasedUserCrnProvider.getAccountId();
+        return freeIpaSecretRotationService.rotateSecretsByCrn(accountId, environmentCrn, request);
```

Comment:
accountid can be extracted from environmentcrn

Reply:
Sure. I'll change it.